pkg/antctl/command_definition.go

```diff
@@ -98,17 +114,17 @@ func (e *resourceEndpoint) flags() []flagInfo {
 }
 
 type nonResourceEndpoint struct {
-	path     string
-	params   []flagInfo
-	isSingle bool
+	path       string
+	params     []flagInfo
+	outputType OutputType
```

Comment:
Have a question here: for registered APIGroup exposed via API Server, could we have the "single" case, but is the API path exposes a single object, but not a type of resources?
I am asking because I am adding the controller-info cmd which should be exposed via aggregated API, and wondering it should be declared as a resourceEndpoint or nonResourceEndpoint in antctl.go. The former does not have the "single" type now,
@weiqiangt @tnqn

Reply:
I think the `controller-info` is a similar case of `version`.
If you specified a resourceName, the antctl would  always assume the response is single. The antctl does not know what the content is but just dispatch the response to the transform function according to definition.

Reply:
Sure. My question is could resourceEndpoint  also have the "single" type case, or it should be always "multiple"?

Reply:
For now, if a resourceEndpoint specified `resourceName` then it is single inexplicitly, otherwise, the antctl will only assume the response is single when running command with arg.
I don't think @lzhecheng implemented multiple for resourceEndpoint, but developers can make the both single transform func and list transform func expect same type to hack a multiple behavior.